packages/plugin-odoo-snippets/src/OdooSnippet.ts

```diff
@@ -297,6 +317,30 @@ export class OdooSnippet<T extends JWPluginConfig = JWPluginConfig> extends JWPl
         const cover = covers && covers[0];
         if (cover) return domLayout.getDomNodes(cover)[0];
     }
+    async getSelectedText(params: CommandParams): Promise<string> {
+        const selectedNode = params.context.range.selectedNodes().map(node => node.clone());
+        const container = new ContainerNode();
+        container.append(...selectedNode);
+        const renderer = this.editor.plugins.get(Renderer);
+        const renderedNode = await renderer.render<HTMLElement[]>(
+            HtmlDomRenderingEngine.id,
+            container,
+        );
+        // todo: what if there is more rendered nodes?
+        return renderedNode ? renderedNode[0].innerText : '';
+    }
+    async getSelectedLink(params: CommandParams): Promise<string> {
+        const selectedNode = params.context.range.selectedNodes().map(node => node.clone());
+        const nodeWithFormat: InlineNode = selectedNode.find(
+            node =>
+                node instanceof InlineNode &&
```

Comment:
Never do `instanceof` for vNodes, we have `is` for that...

Reply:
To be fair, we discussed with chm a few days ago and he told me that using `instanceof` would be orders of magnitude faster than using `.is` once the memory is merged. I'm not feeling confident about the usefulness of `.is` apart from being shorter to write so it's possible it might get the axe at some point.